NBitcoin/Protocol/NodeServer.cs

```diff
@@ -179,32 +179,34 @@ private void EndAccept(SocketAsyncEventArgs args)
 					if(_Cancel.IsCancellationRequested)
 						return;
 					NodeServerTrace.Information("Client connection accepted : " + client.RemoteEndPoint);
-					var cancel = CancellationTokenSource.CreateLinkedTokenSource(_Cancel.Token);
-					cancel.CancelAfter(TimeSpan.FromSeconds(10));
-
-					var stream = new NetworkStream(client, false);
-					while(true)
+					using(var cancel = CancellationTokenSource.CreateLinkedTokenSource(_Cancel.Token))
 					{
-						if(ConnectedNodes.Count >= MaxConnections)
+						cancel.CancelAfter(TimeSpan.FromSeconds(10));
+
+						var stream = new NetworkStream(client, false);
+						while(true)
 						{
-							NodeServerTrace.Information("MaxConnections limit reached");
-							Utils.SafeCloseSocket(client);
-							break;
+							if(ConnectedNodes.Count >= MaxConnections)
+							{
+								NodeServerTrace.Information("MaxConnections limit reached");
+								Utils.SafeCloseSocket(client);
+								break;
+							}
+							cancel.Token.ThrowIfCancellationRequested();
+							PerformanceCounter counter;
+							var message = Message.ReadNext(stream, Network, Version, cancel.Token, out counter);
+							_MessageProducer.PushMessage(new IncomingMessage()
+							{
+								Socket = client,
+								Message = message,
+								Length = counter.ReadenBytes,
+								Node = null,
+							});
+							if(message.Payload is VersionPayload)
+								break;
+							else
+								NodeServerTrace.Error("The first message of the remote peer did not contained a Version payload", null);
 						}
-						cancel.Token.ThrowIfCancellationRequested();
-						PerformanceCounter counter;
-						var message = Message.ReadNext(stream, Network, Version, cancel.Token, out counter);
-						_MessageProducer.PushMessage(new IncomingMessage()
-						{
-							Socket = client,
-							Message = message,
-							Length = counter.ReadenBytes,
-							Node = null,
-						});
-						if(message.Payload is VersionPayload)
-							break;
-						else
-							NodeServerTrace.Error("The first message of the remote peer did not contained a Version payload", null);
 					}

```

Comment:
Yes, nobody can. There is only one and small change here: the `cancel` cts was wrapped by an `using` and all the code inside it was indented. For some reason the diff tool seems to be confused and display this as a complicated change.

Reply:
I didn't mean I don't understand your code. I meant I don't understand @NicolasDorier's code. If there's something inside that's been kicked off asynchronously, because in that case you cannot dispose with a using there.